gamification/process/update.py

```diff
@@ -0,0 +1,140 @@
+import logging
+
+from data.models import (
+    MergeRequest,
+    Issue,
+)
+from gamification.process.activity_points import (
+    get_activity_with_points
+)
+from gamification.labels import NEGATIVE_POINT_LABELS
+from gamification.data.points import MERGE_REQUEST_CLOSED_WITHOUT_MERGE
+from gamification.models import Participant
+
+
+def get_mr_objects():
+    """
+    Get mrs objects saved in the database.
+
+    :return: a queryset of mr objects.
+    """
+    mrs = MergeRequest.objects.all()
+    return mrs
+
+
+def update_participants_data_with_mr(mr):
+    """
+    Update particiapnts data based on mr.
+
+    This method updates the total score earned by the
+    participant based on the activities performed,
+    and then update the current level based on the
+    total score.
+
+    This method first check if the mr is merged or not,
+    if it's merged, then it's get a activity and points based
+    on the labels on mr and update the participant data who
+    opened this mr.
+
+    Further it gets all the issues that mr is closing then get the
+    points and activity based on the labels on the issue and update
+    the participant data who opened that issue.
+    """
+    logger = logging.getLogger(__name__)
+
+    # Get the participant who opened this mr
+    mr_author = Participant.objects.get(username=mr.author)
+    mr_performed_at = mr.created_at
+    mr_updated_at = mr.updated_at
+
+    if mr.state == 'merged':
+        mr_labels = mr.labels.values('name')
+
+        # Get activity and points based on labels on the mr
+        mr_points, mr_activity_string = get_activity_with_points(
+            'merge_request', mr_labels)
+
+        # Update participant data
+        mr_author.add_points(mr_points,
+                             mr_activity_string,
+                             mr_performed_at,
+                             mr_updated_at,
+                             )
+        # Get all the issues this mr is closing
+        try:
+            issues = mr.get_closes_issues_object()
+        except Exception as ex:
+            logger.error(ex)
+            return
+        for issue in issues:
+            issue_labels = issue.labels.values('name')
+            issue_performed_at = issue.created_at
+            issue_updated_at = issue.updated_at
+            # Get activity and points based on the labels on the issue
+            issue_points, issue_activity_string = get_activity_with_points(
+                'issue', issue_labels)
+            if issue.author == mr.author:
+                issue_author = mr_author
```

Comment:
`mr_author` is invalid.

and the fact that this didnt cause a build break worries me.

does that mean we have zero mrs (by newcomers) where the MR author is the same as the issue author?

I might be wrong, but my guess is that at least @harshhx17 has created an issue and also worked on it.

Reply:
ahhh .. `issue.author == mr.author` and `issue_author = mr_author` is confusing because of the `.` vs `_`.

Use `if issue.author == mr_author.username:` ?

Reply:
Strange that we are not getting coverage when we use `if issue.author == mr_author.username:` but `issue.author == mr.author` the full coverage.

Reply:
ok, not worth the pain of debugging this atm